ci_watson/artifactory_helpers.py

```diff
@@ -88,6 +90,16 @@ def get_bigdata(*args):
     args : tuple of str
         Location of file relative to ``TEST_BIGDATA``.
 
+    repo : str
+        Name of repository on Artifactory where data is located.  This will be
+        appended to whatever was specified in ``TEST_BIGDATA`` to complete
+        the URL for finding the data for the test, if ``TEST_BIGDATA`` was
+        specified as a URL in the first place. Default: ''
+
+    copy_local : bool
```

Comment:
`copy_local` is too confusing for me in the way it is used here because basically `_download(src, dst)` is the same as `copy_local` (i.e., getting that file to working directory) but the latter is not controlled by this option.

I propose an alternative implementation that is something like `no_copy` (default is `False`). And its behaviour should "stick" regardless whether path is URL or not (because you never know if the code you are testing is able to stream directly from URL or not).

Reply:
Renaming this option to `no_copy` would be fine with me.  I will look into making the logic for this parameter consistent with that name. 

As for that behavior, the dual nature of this implementation is something which we have no choice about.  Anytime a URL is provided, the data has to be copied to a local directory, as I am unaware of any FITS interface that performs a read-only operation over a URL.  Also, I am unaware of any ability of the JWST datamodels code being able to open an input file across a URL.  These limitations lock us into ALWAYS performing a copy with a URL regardless of this setting.  If the file is already on a locally accessible location, then this parameter can direct the code to simply read from that location rather than performing an unnecessary copy.

Reply:
I not not like this idea.